src/cpu/reorder/simple_reorder.hpp

```diff
@@ -198,8 +202,8 @@ struct simple_reorder_impl<SIMPLE_REORDER_TEMPL_CALL,
                 : 1.f;
 
         size_t offset = output_d.size() - output_d.additional_buffer_size();
-        size_t zp_offset = offset
-                + (req_comp ? G * dims[w_groups + 0] * sizeof(int32_t) : 0);
+        size_t zp_offset
+                = offset + (req_comp ? G * PADDED_OC * sizeof(int32_t) : 0);
```

Comment:
I would suggest to use a bit more scalable solution (as it seems to me), make `additional_buffer_size()` function to accept an argument (which will be `extra().flags` by default), that chooses the specific comp type. In this case it will be `s8s8_comp`.
Then `zp_offset` can be computed as `output_d.size() - output_d.additional_buffer_size(s8s8_comp)`. No need to follow the padding values, dimensions, etc.
This implicitly relies on the order of compensations, but we still do it any way.

Reply:
Makes sense, I added 2 commits to reuse `additional_buffer_size()` for offset calculations.